src/mjolnir/landmarks.cc

```diff
@@ -296,6 +318,226 @@ bool BuildLandmarkFromPBF(const boost::property_tree::ptree& pt,
+    }
+
+    if (!tile_builder_ptr ||
+        tile_builder_ptr->header_builder().graphid().Tile_Base() != (*it).first.Tile_Base()) {
+      // store the previously updated tile
+      if (tile_builder_ptr) {
+        tile_builder_ptr->StoreTileData();
+        updated_tiles++;
+      }
+      // reset the tile builder ptr to the current tile
+      tile_builder_ptr.reset(new GraphTileBuilder(tile_dir, (*it).first.Tile_Base(), true));
+    }
+
+    // retrieve the landmark to be added
+    const std::vector<Landmark> landmark =
+        db.get_landmarks_by_ids({static_cast<int64_t>((*it).second)});
+    if (landmark.size() != 1) {
+      throw std::logic_error("Incorrect result size " + std::to_string(landmark.size()) +
+                             " of retrieved landmarks, which should be 1");
+    }
+    // add the landmark to the tile
+    GraphId edge_id = (*it).first;
+    tile_builder_ptr->AddLandmark(edge_id, landmark[0]);
+
+    // update the counting numbers
+    updated_landmarks++;
+    if (last_edge != (*it).first) {
+      updated_edges++;
+      last_edge = (*it).first;
+    }
+  }
+  // store the last updated tile
+  if (tile_builder_ptr) {
+    tile_builder_ptr->StoreTileData();
+    updated_tiles++;
+  }
+
+  // set the stats
+  stats.set_value(std::make_tuple(updated_tiles, updated_edges, updated_landmarks));
+}
+
+// Add all landmarks to tiles
+bool AddLandmarks(const boost::property_tree::ptree& pt) {
+  LOG_INFO("Starting adding landmarks to tiles...");
+
+  const size_t num_threads =
+      pt.get<size_t>("mjolnir.concurrency", std::thread::hardware_concurrency());
+  const std::string db_name = pt.get_child("mjolnir").get<std::string>("landmarks_db", "");
+
+  // get tile access
+  baldr::GraphReader reader(pt.get_child("mjolnir"));
+
+  // get all tile ids and sort the tiles in descending order by size to balance the threads
+  auto tileset = reader.GetTileSet(1);
+  std::vector<GraphId> vec_tileset(tileset.begin(), tileset.end()); // turn into a vector for sorting
+
+  std::sort(vec_tileset.begin(), vec_tileset.end(), [&](const auto id_a, const auto id_b) {
+    return reader.GetGraphTile(id_a)->header()->nodecount() >
+           reader.GetGraphTile(id_b)->header()->nodecount();
+  });
+
+  LOG_INFO("Finding landmarks and their correlated edges...");
+
+  std::vector<std::shared_ptr<std::thread>> threads(num_threads);
+  std::vector<std::promise<std::string>> sequence_file_names(num_threads);
+  for (size_t i = 0; i < num_threads; ++i) {
+    threads[i].reset(new std::thread(FindLandmarkEdges, std::cref(pt), std::cref(vec_tileset), i,
+                                     num_threads, std::ref(sequence_file_names[i])));
+  }
+
+  // join all the threads and collect the sequence file names
+  for (auto& thread : threads) {
+    thread->join();
+  }
+
+  std::vector<std::string> seq_names{};
+  for (std::promise<std::string>& s : sequence_file_names) {
+    seq_names.push_back(s.get_future().get());
+  }
+
```

Comment:
```suggestion

  LOG_INFO("Sorting landmark edge pairs by tile...");

```

Reply:
ah finally!!! seems github preserves comments in file view once they were addressed. was the most annoying UX oversight IMO..